internal/character/seele/talent.go

```diff
@@ -0,0 +1,70 @@
+package seele
+
+import (
+	"github.com/simimpact/srsim/pkg/engine/event"
+	"github.com/simimpact/srsim/pkg/engine/info"
+	"github.com/simimpact/srsim/pkg/engine/modifier"
+	"github.com/simimpact/srsim/pkg/engine/prop"
+	"github.com/simimpact/srsim/pkg/key"
+	"github.com/simimpact/srsim/pkg/model"
+)
+
+const (
+	Resurgence key.Modifier = "seele-resurgence"
+)
+
+// Enters the buffed state upon defeating an enemy with Basic ATK, Skill, or Ultimate,
+// and receives an extra turn. While in the buffed state,
+// the DMG of Seele's attacks increases by 80% for 1 turn(s).
+// Enemies defeated in the extra turn provided by "Resurgence"
+// will not trigger another "Resurgence."
+
+func init() {
+	modifier.Register(Resurgence, modifier.Config{
+		Stacking:   modifier.ReplaceBySource,
+		StatusType: model.StatusType_STATUS_BUFF,
+	})
+}
+
+func (c *char) initTalent() {
+	// if seele kills in her turn, set extra turn trigger and add buffedState mod.
+	c.engine.Events().TargetDeath.Subscribe(func(e event.TargetDeath) {
+		if e.Killer == c.id && c.id == c.actionOwner {
+			c.shouldInsert = true
+			c.enterBuffedState()
+		}
+	})
+
+	// actionStart subs to enable adding buffedState mod effectively after-kill
+	c.engine.Events().ActionStart.Subscribe(func(e event.ActionStart) {
+		if !e.IsInsert {
+			c.actionOwner = e.Owner
+		}
+	})
+
+	// extra turn logic
+	c.engine.Events().ActionEnd.Subscribe(func(e event.ActionEnd) {
+		if c.shouldInsert {
+			c.shouldInsert = false
+			c.engine.InsertAction(c.id)
+		}
+		c.shouldInsert = false
+	})
```

Comment:
nit: `shouldInsert` being set false twice is redundant. Delete case inside if statement.